packages/schematics/angular/migrations/update-6/index_spec.ts

```diff
@@ -0,0 +1,638 @@
+        });
+
+        it('should not move spec if not defined', () => {
+          tree.create(oldConfigPath, JSON.stringify(baseConfig, null, 2));
+          tree = schematicRunner.runSchematic('migration-01', defaultOptions, tree);
+          const config = getSchematicConfig(tree, 'service');
+          expect(config).toBeUndefined();
+        });
+      });
+    });
+
+    describe('architect', () => {
+      it('should exist', () => {
+        tree.create(oldConfigPath, JSON.stringify(baseConfig, null, 2));
+        tree = schematicRunner.runSchematic('migration-01', defaultOptions, tree);
+        const config = getConfig(tree);
+        expect(config.architect).not.toBeDefined();
+      });
+    });
+
+    describe('app projects', () => {
+      it('should create two projects per app', () => {
+        tree.create(oldConfigPath, JSON.stringify(baseConfig, null, 2));
+        tree = schematicRunner.runSchematic('migration-01', defaultOptions, tree);
+        const config = getConfig(tree);
+        expect(Object.keys(config.projects).length).toEqual(2);
+      });
+
+      it('should create two projects per app', () => {
+        baseConfig.apps.push(baseConfig.apps[0]);
+        tree.create(oldConfigPath, JSON.stringify(baseConfig, null, 2));
+        tree = schematicRunner.runSchematic('migration-01', defaultOptions, tree);
+        const config = getConfig(tree);
+        expect(Object.keys(config.projects).length).toEqual(4);
+      });
+
+      it('should use the app name if defined', () => {
+        baseConfig.apps[0].name = 'foo';
+        baseConfig.apps.push(baseConfig.apps[0]);
+        tree.create(oldConfigPath, JSON.stringify(baseConfig, null, 2));
+        tree = schematicRunner.runSchematic('migration-01', defaultOptions, tree);
+        const config = getConfig(tree);
+        expect(config.projects.foo).toBeDefined();
+        expect(config.projects['foo-e2e']).toBeDefined();
+      });
+
+      it('should set the project root values', () => {
+        tree.create(oldConfigPath, JSON.stringify(baseConfig, null, 2));
+        tree = schematicRunner.runSchematic('migration-01', defaultOptions, tree);
+        const project = getConfig(tree).projects['app1'];
+        expect(project.root).toEqual('src');
+        expect(project.projectType).toEqual('application');
+      });
+
+      it('should set build target', () => {
+        tree.create(oldConfigPath, JSON.stringify(baseConfig, null, 2));
+        tree = schematicRunner.runSchematic('migration-01', defaultOptions, tree);
+        const build = getConfig(tree).projects['app1'].architect.build;
+        expect(build.builder).toEqual('@angular-devkit/build-webpack:browser');
+        expect(build.options.scripts).toEqual([]);
+        expect(build.options.styles).toEqual([{ input: 'src/styles.css' }]);
+        expect(build.options.assets).toEqual([{ glob: 'src/assets' }, { glob: 'src/favicon.ico' }]);
+        const prodConfig = build.configurations.production;
```

Comment:
Note: you can actually use Jasmine's `toEqual` to compare objects, and it will tell you if there are missing/extra members. 

I did it in https://github.com/angular/angular-cli/blob/5716621e3e5d42007c4109201c03705c49e445b5/packages/%40angular/cli/utilities/build-webpack-compat.spec.ts#L234-L242. 

Not a change request, just a comment.